docs/service-overview.md

```diff
@@ -0,0 +1,947 @@
+
+## Service Authentication
+
+### Authentication
+
+It is assumed that the vendor providing this service endpoint has a mechanism to
+provide authentication for the service endpoint.  The following examples assume
+the use of an OIDC-based authentication environment.
+
+```mermaid
+flowchart TB
+
+    orgBlock["Org: AcmeCo
+    Org ID: org-acmeco
+    Users = [{username = admin, role = admin}]"]
+
+    topGroup("Group: Default
+    Group ID: group-e51b5c2c-eda1-4c27-8a86-ece7faab33f
+    Domain Certs = [&lt;x509 domain cert 1&gt;, &lt;x509 domain cert 2&gt;]
+    Switch Serials = [JGEXXXXXX]
+    Users = [{username = useracm, role = requestor}]")
+
+    siteA("Group: SiteA
+    Group ID: group-ad636092-fbc4-446c-ab2f-9df16719613a
+    Switch Serials = [GACXXXXXX]
+    Users: [{username = userconsulting, role = admin}]")
+
+    siteB("
+    Group: SiteB
+    Group ID: group-79ef6ff1-98d1-4fa7-98f3-7d98af4ee6ee
+    Users: [{username = siteb, role = requestor}]")
+
+    orgBlock --> topGroup
+    topGroup --> siteA
+    topGroup --> siteB
+```
+
+### Authorization
+
+The above figure shows the relationships between different entities in the
+system. This is the organization tree. The root of the tree is the organization
+itself. In this example, the organization has one child group, which has two
+child groups. Each group has a set of pinned domain certs, a set of users, and a
+set of serials assigned to it. This system has 3 roles: **ADMIN**, **ASSIGNER**,
+and **REQUESTOR**. An **ADMIN** can do everything that an **ASSIGNER** can do.
+An **ASSIGNER** can do everything a **REQUESTOR** can do.
+
+Each user has access to the group that it belongs to along with all of its
+children, i.e., the user has access to the subtree rooted at the group that the
+user belongs to. One user can belong to multiple groups and access multiple
+subtrees in the hierarchy. The user has the same permission across the subtree
+that it has been assigned at the root group of the subtree. If two subtrees
+overlap for a user, the permission that gives the user higher access is applied.
+
+In the example shown in the figure, the user with `username=siteb` cannot request
+an ownership voucher for the serial JGEXXXXXX because the serial belongs to the
+parent group. The user with username=useracm can request an ownership voucher
+for the serial with serial number GACXXXXXX because the user has access to the
+parent group of the serial.
+
+## Serial Numbers
+
+Ownership vouchers can be issued for network elements as identified by their
+serial number. Some products may also have a TPM Public Key associated with
+them. By default, it is assumed that all the customer-owned serials are added to
+the root group. Users with the **ASSIGNER** or **ADMIN** role can move/add
+serials within the tree from there. A serial can be assigned to a single group
+apart from the root group. See the `/GetSerial` RPC for an example of the
+details available for a given serial number.
```

Comment:
We'd like to mention that by default, serials for all elements/components owned by a specific org are added to the Root Group, which is precreated and named as org_id

Reply:
I see this is mentioned below. Upto you if you want to add this bit of detail here too or simply ignore this comment.